File: Document/Python-style pseudocode document/urana_process/fittable_mapper/nn/original_cnn/containers/io/value/cnn_output_vector.py
class CnnOutputVector:
    def __init__(self, size_or_vector):
        if isinstance(size_or_vector, int):
            self.underlying = FloatVector(size_or_vector)
        else:
            self.underlying = size_or_vector

    def getVector(self):
        return self.underlying

    def close(self): ...


from core import FloatVector

File: src/main/java/com/github/magif1712/smarter_touhou_maids/features/smarter/agent/reflex_arc_system_agent/ai/process_ai/process/urana_process/fittable_mapper/nn/cnn/containers/io/value/CnnOutputVector.java
package com.github.magif1712.smarter_touhou_maids.features.smarter.agent.reflex_arc_system_agent.ai.process_ai.process.urana_process.fittable_mapper.nn.cnn.containers.io.value;

import com.github.magif1712.smarter_touhou_maids.core.containers.vector.FloatVector;

/**
 * CNN 输出向量：双构造——{@code int} 时新建 {@link FloatVector}，{@code FloatVector} 时直接包。
 * {@code close()} 真实释放 underlying。
 */
public class CnnOutputVector implements AutoCloseable {
    private final FloatVector underlying;

    public CnnOutputVector(int size) {
        this.underlying = new FloatVector(size);
    }

    public CnnOutputVector(FloatVector vector) {
        this.underlying = vector;
    }

    public FloatVector getVector() {
        return underlying;
    }

    @Override
    public void close() {
        underlying.close();
    }
}